### ISSUE TITLE
can't git pull from bare repository

### ISSUE THREAD

**Issue by domanchi** [Yelp/detect-secrets-server#21]
Wow. I'm so surprised that this *only* appeared now. It seems like such a critical bug.

When pulling master from a bare repository, it's going to fail with something like:

```
error: Your local changes to the following files would be overwritten by merge:
        CHANGELOG.md
        README.md
        src/OutsideClickHandler.jsx
        test/OutsideClickHandler_test.jsx
Please commit your changes or stash them before you merge.
Aborting
```

This is because bare repositories don't have the files in them, and pulling *pulls* those files into your local copy.

Instead, what we need to do is:

```
git fetch origin master:master
```

This will pull the changes, but not the files themselves.

**Comment by KevinHock** [comment]
Just summarizing the historical stuff:

We originally did:

```python
try:
    # Retrieve the current branch name
    main_branch = subprocess.check_output([
        'git',
        '--git-dir',
        repo_location,
        'rev-parse',
        '--abbrev-ref',
        'HEAD'
    ], stderr=subprocess.STDOUT).strip()

    # Fetch the latest HEAD into the bare repo
    subprocess.check_output([
        'git',
        '--git-dir',
        repo_location,
        'fetch',
        '-q',
        'origin',
        main_branch
    ], stderr=subprocess.STDOUT)
except subprocess.CalledProcessError as e:
    error_msg = e.output.decode('ascii')
    print(error_msg)
    return
``` 

then changed it to 

```python
    subprocess.check_output([
        'git',
        '--git-dir', self.repo_location,
        '--work-tree', '.',
        'pull',
    ], stderr=subprocess.STDOUT)
```

and don't remember why. We're gonna change it back to what it was originally.

**Comment by KevinHock** [comment]
Thanks to #23 this should now be fixed in `0.2.7` https://github.com/Yelp/detect-secrets-server/blob/master/CHANGELOG.md#027
